internal/exec/selected/selected.go

```diff
@@ -56,16 +70,86 @@ type SchemaField struct {
 	FixedResult reflect.Value
 }
 
+func (f *SchemaField) Kind() selected.Kind {
+	return selected.FieldKind
+}
+
+func (f *SchemaField) Identifier() string {
```

Comment:
Couldn't use `Name()` method here yet since the backing `internal/exec/selected/*SchemaField` already has a field `Name` so it conflicts. 

At one point, if we all agree on the feature and the interface, I suggest the internal field changes to the interface looks better.

This is the same reasoning for `Aliased` below.

Reply:
Why can't you use its own `Name` field then?

Reply:
Because the new public interface would exposes it, hence a field with the same name cannot exist.